File: CPP/horner.cpp
#include <iostream>

int main(int argc, char **argv)
{
	int argument;
	cout << "Podaj argument";
	cin >> argument;
	int stopien;
	cout << "Podaj stopien";
	cin >> stopien;
	int n;
	n = stopien + 1;
	int wynik;
	int wsp[n];
	for(int i=0;i < n; i++){
	    cout << "Podaj współczynnik:";
	    cin >> tab[i];
	}
	for(int j = 0; j < stopien + 1; j++){
	    if(j == 0){
	        wynik = wsp[i];
	    }
	    else{
	        wynik = argument * wynik + wsp[i];
	    }
	}
    
    cout << wynik;	
}

File: python/horner.py
def homer_it(stopien,wsp,argument):
	for i in range(stopien+1):
		if i == 0:
			wynik = wsp[i]
		else:
			wynik = argument * wynik + wsp[i]
	print("Wynik:",wynik)

def main(args):
	stopien = int(input("Podaj stopień wielomianu:"))
	argument = int(input("Podaj argument:"))
	wsp = []
	for i in range(stopien+1):
		a = int(input("Podaj współczynnik:"))
		wsp.append(a)
	homer_it(stopien,wsp,argument)
	
	
	return 0
	
if __name__ == '__main__':
	import sys
	sys.exit(main(sys.argv))
